experiments/009_probablistic/readme.md

```diff
@@ -1,15 +1,48 @@
 # Probablistic Forecasts with Quantile Regression
 
 Lets investigate the effects of quantil regression on the results. The deciles of 10% and 90% are also forecasted.
-Additionally, the MAE of the median is calculated as well.
+Additionally, the MAE of the median is calculated as well. This is the Pinball loss from scipy for the quantiles. 
+This is supposed to be the same as MAE when the quantile is 0.5, but seems to be slightly different for this case, about half the level of 
+our MAE for the median.
 
 | Horizon | 0.1   | 0.5   | 0.9   | MAE 0.5 | Prev MAE 0.5 |
 |---------|-------|-------|-------|---------|---------|
-| 0       | 0.016 | 0.010 | 0.009 | 0.009   | 0.009 |
-| 1       | 0.021 | 0.016 | 0.014 | 0.014   | 0.014 |
-| 2       | 0.024 | 0.018 | 0.017 | 0.016   | 0.017 |
-| 4       | 0.025 | 0.020 | 0.019 | 0.018   | 0.019 |
-| 8       | 0.026 | 0.020 | 0.019 | 0.019   | 0.019 |
-| 16      | 0.025 | 0.020 | 0.018 | 0.018   | 0.018 |
-| 24      | 0.025 | 0.020 | 0.019 | 0.018   | 0.019 |
-| 36      | 0.026 | 0.021 | 0.019 | 0.019   | 0.019 |
+| 0       | 0.00261 | 0.00444 | 0.00245 | 0.009   | 0.009 |
+| 1       | 0.0041 | 0.00686 | 0.00377 | 0.014   | 0.014 |
+| 2       | 0.00526 | 0.00858 | 0.00471 | 0.016   | 0.017 |
+| 4       | 0.00543 | 0.00879 | 0.00486 | 0.018   | 0.019 |
+| 8       | 0.0054 | 0.00897 | 0.00531 | 0.019   | 0.019 |
+| 16      | 0.00547 | 0.00886 | 0.00509 | 0.018   | 0.018 |
+| 24      | 0.00518 | 0.00886 | 0.00525 | 0.018   | 0.019 |
+| 36      | 0.00526 | 0.00929 | 0.00556 | 0.019   | 0.019 |
+
+Counts below the quantiles. These show the number of times the actual value was below the quantile.
+For 10th percentile, it should be around 10%, and for 90% it should be around 90%. It seems that the model
+is not necessarily fully calibrated. This is also done for non-night time (defined as true generation above 1% of the capacity), and
+for all times, including night time.
+
+All times Percentiles:
```

Comment:
@dantravers The percentile counts are here now. It is kinda odd, for the test set, when looking at all forecast times, the 90th percentile is larger than ~86-88% of the forecasts, so fairly close, but the 10th percentile is also larger than 60-64% of the actual values. But, when only looking at times where the true generation is at least 1% (0.01), to cut out night times, then the 10th percentile is larger than 12-15% of the forecasts, so still too large, but closeish, but then the 90th percentile is only larger than 35-37% of the generation. These results kinda confuse me, and not really sure what they mean. I would have expected maybe including all times could skew the percentages as night time is just 0, but didn't expect it to be such wide swings, and only one of the percentiles being close in each case.

Reply:
@peterdudfield I've added some percentile counts, as above, results seem a bit odd. This is all on the 2021 test data.

Reply:
hmm, this seems a bit odd, do you see this in the plots too?

Reply:
Agree with you choice to remove night times.  But the results for the 90th percentile are wierd - according to this, the 90th percentile is not even as bit as the median!    I agree - worth looking at some charts - maybe just the timeseries of the 90% boundary & the actuals...

Reply:
In the plots that I had before, it doesn't look this odd, and looks roughly right in my eyes. I haven't tried plotting the whole of the 2021 though to see. But yeah, can make some more plots to check it more, I'll make a few of a couple days at a time of the different percentile actuals, and with and without night time.

Reply:
I added some plots and the code I used to generate these numbers here: https://github.com/openclimatefix/uk-pv-national-xg/issues/66#issuecomment-1545253444

Reply:
Ah, wait, I found a bug, I have been dividing the daytime-only ones by the wrong number, the total number of points, including nighttime ones

Reply:
These numbers will be updated, and for the daytime only ones it gets closer to making sense. On daylight-only values, the 10th percentile is larger than around 28% of values, the 50th around 57%, and the 90th larger than 76% of values. So the outputs are still not as wide as we would expect, but a lot closer to what it should be.

Reply:
Running it on the train set is being added here: https://github.com/openclimatefix/uk-pv-national-xg/issues/66#issuecomment-1545359963 where the quantiles seems to be quite close to where they should be

Reply:
Hmm - the quantiles are a lot narrower than we'd expect (like almost half of what they should be!)  Any differences between the test and train set?  Maybe swap them around and see if get the same effect?  Good discussion for ML meeting...